### PR TITLE
Fix ApiManager not handling duplicate apiBases of KubeApis

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

### CODE REVIEW

File: packages/core/src/common/k8s-api/api-manager/api-manager.ts [hunk 1]
```diff
@@ -41,19 +41,22 @@ export class ApiManager {
       const apis = chain(this.dependencies.apis.get().values())
         .concat(this.externalApis.values());
       const removedApis = new Set(this.apis.values());
+      const newState = new Map(this.apis);
```
Comment (panuhorsmalahti) [inline packages/core/src/common/k8s-api/api-manager/api-manager.ts:44]: [nit] `const newApis`

Comment (Nokel81) [reply packages/core/src/common/k8s-api/api-manager/api-manager.ts:44]: Technically it is also `oldApis` since we don't necessarily remove all of them.

**Review by jakolehm — APPROVED** [review]
LGTM

File: packages/core/src/common/k8s-api/__tests__/api-manager.test.ts [hunk 2]
```diff
@@ -72,21 +75,48 @@ describe("ApiManager", () => {
...
       Object.defineProperty(kubeApi, "apiBase", { value: fallbackApiBase });
-      apiManager.registerApi(fallbackApiBase, kubeApi);
+      apiManager.registerApi(kubeApi);
 
       // Test that store is returned with new apiBase
       expect(apiManager.getStore(kubeApi)).toBe(kubeStore);
     });
   });
+
+  describe("technical tests for autorun", () => {
```
Comment (jim-docker) [inline packages/core/src/common/k8s-api/__tests__/api-manager.test.ts:96]: could add another test where the second api is removed and then `apiManager.getApi(apiBase)` returns `firstApi`?

**Review by jim-docker — APPROVED** [review]

File: packages/core/src/common/k8s-api/__tests__/api-manager.test.ts (hunk 2 above)
Comment (Nokel81) [reply packages/core/src/common/k8s-api/__tests__/api-manager.test.ts:96]: Good idea

Comment (Nokel81) [reply packages/core/src/common/k8s-api/__tests__/api-manager.test.ts:96]: Actually this is not possible given that the apis are registered automatically and then never garbage collected.

**Pull request merged**